internal/monitors/elasticsearch/client/client.go

```diff
@@ -35,44 +43,43 @@ func NewESClient(host string, port string, scheme string, username string, passw
 }
 
 // Method to fetch node stats
-func (c *esClient) GetNodeAndThreadPoolStats(result *NodeStatsOutput) error {
+func (c *esClient) GetNodeAndThreadPoolStats() (*NodeStatsOutput, string, error) {
 	url := fmt.Sprintf("%s://%s:%s/%s", c.scheme, c.host, c.port, nodeStatsEndpoint)
 
 	body, err := get(url, c.username, c.password, *c.httpClient)
 
 	if err != nil {
-		return err
+		return nil, url, err
```

Comment:
Instead of returning the URL up the call stack, just wrap the error message so that it includes the URL.  That way, in your [main RunOnInterval function](https://github.com/signalfx/signalfx-agent/pull/648/files/0dd9b359126222ad238e10d8b9ad8dc99fb58f5d..bc625c7fa992f2b96d8460d98e54fd6226fd3185#diff-d839d4d48563fddb8b5872435d15c20bR72), you don't need to actually deal directly with the URL.

So like here it would be like: `return nil, fmt.Errorf("could not get url %s: %v", url, err)`.

Sorry, didn't realize the URL wasn't readily available in that caller.

Reply:
Moved this to the method that actually fetches the URL